subprojects/packagefiles/openssl/meson.build

```diff
@@ -55,88 +55,124 @@ else
-    arch_subdir = 'linux-aarch64'
-  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
-    arch_subdir = 'BSD-x86'
-  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['linux', 'android']
-    arch_subdir = 'linux-elf'
-  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'darwin'
-    arch_subdir = 'darwin-i386-cc'
-  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'sunos'
-    arch_subdir = 'solaris-x86-gcc'
-  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'windows'
-    warning('x86 + windows combo does not support ASM yet, please contribute')
-    asm = 'no-asm'
-    # TODO: Port this for Windows and uncomment
-    # arch_subdir = 'VC-WIN32'
-    # asm = 'asm'
-    #'rules': [
-    #  {
-    #    'rule_name': 'Assemble',
-    #    'extension': 'asm',
-    #    'inputs': [],
-    #    'outputs': [
-    #      '<(INTERMEDIATE_DIR)/<(RULE_INPUT_ROOT).obj',
-    #    ],
-    #    'action': [
-    #      'nasm.exe',
-    #      '-f win32',
-    #      '-o', '<(INTERMEDIATE_DIR)/<(RULE_INPUT_ROOT).obj',
-    #      '<(RULE_INPUT_PATH)',
-    #    ],
-    #  }
-    #],
-  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
-    arch_subdir = 'BSD-x86_64'
-  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'darwin'
-    arch_subdir = 'darwin64-x86_64-cc'
-  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() == 'darwin'
-    arch_subdir = 'darwin64-arm64-cc'
-  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'sunos'
-    arch_subdir = 'solaris64-x86_64-gcc'
-  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'windows'
-    warning('x86_64 + windows combo does not support ASM yet, please contribute')
-    asm = 'no-asm'
-    # TODO: Port this for Windows and uncomment
-    # arch_subdir = 'VC-WIN64A'
-    # asm = 'asm'
-    #'rules': [
-    #  {
-    #    'rule_name': 'Assemble',
-    #    'extension': 'asm',
-    #    'inputs': [],
-    #    'outputs': [
-    #      '<(INTERMEDIATE_DIR)/<(RULE_INPUT_ROOT).obj',
-    #    ],
-    #    'action': [
-    #      'nasm.exe',
-    #      '-f win64',
-    #      '-DNEAR',
-    #      '-Ox',
-    #      '-g',
-    #      '-o', '<(INTERMEDIATE_DIR)/<(RULE_INPUT_ROOT).obj',
-    #      '<(RULE_INPUT_PATH)',
-    #    ],
-    #  }
-    #],
-  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() in ['linux', 'android']
-    arch_subdir = 'linux-x86_64'
-  elif host_machine.cpu_family() == 'mips64' and host_machine.system() == 'linux'
-    arch_subdir = 'linux64-mips64'
+  if is_bsd
+    if is_x86
+      arch_subdir = 'BSD-x86'
+    elif is_x86_64
+      arch_subdir = 'BSD-x86_64'
+    else
+      asm = 'no-asm'
+    endif
```

Comment:
In general, these `if os; if arch` style checks with repetitive else clauses seem like maybe they'd actually be less repetitive if we do `if os and arch` style checks... just simplifying the `host_machine` property comparison checks if you like.